src/Controller/EnMarche/InteractiveController.php

```diff
@@ -16,18 +17,17 @@ class InteractiveController extends Controller
      * @Route("/ton-pouvoir-achat", name="app_purchasing_power")
      * @Method("GET|POST")
      */
-    public function purchasingPowerAction(Request $request): Response
+    public function purchasingPowerAction(Request $request, PurchasingPowerProcessorHandler $handler): Response
     {
         $session = $request->getSession();
-        $handler = $this->get('app.interactive.purchasing_power_processor_handler');
         $purchasingPower = $handler->start($session);
         $transition = $handler->getCurrentTransition($purchasingPower);
 
         $form = $this->createForm(PurchasingPowerType::class, $purchasingPower, ['transition' => $transition]);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            if ($purchasingPowerLog = $this->get('app.interactive.purchasing_power_processor_handler')->process($session, $purchasingPower)) {
+            if ($purchasingPowerLog = $handler->process($session, $purchasingPower)) {
```

Comment:
To be conistent, you should use `$this->get(PurchasingPowerProcessorHandler:class)` here and not inject the handler. (it is used in a condition)

Reply:
it's not really the same case, here the variable `$handler` is used on top of the method, L23

Reply:
I did not seen this call, ok for me :+1: